http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/monitor/GKEInterpreter.scala

```diff
@@ -0,0 +1,390 @@
+package org.broadinstitute.dsde.workbench.leonardo
+package monitor
+
+import _root_.io.chrisdavenport.log4cats.StructuredLogger
+import cats.Parallel
+import cats.effect.{Blocker, ConcurrentEffect, ContextShift, Timer}
+import cats.implicits._
+import cats.mtl.ApplicativeAsk
+import com.google.container.v1.MasterAuthorizedNetworksConfig.CidrBlock
+import com.google.container.v1._
+import org.broadinstitute.dsde.workbench.DoneCheckableInstances._
+import org.broadinstitute.dsde.workbench.DoneCheckableSyntax._
+import org.broadinstitute.dsde.workbench.google2.GKEModels._
+import org.broadinstitute.dsde.workbench.google2.KubernetesClusterNotFoundException
+import org.broadinstitute.dsde.workbench.google2.KubernetesModels.{
+  KubernetesNamespace,
+  KubernetesSecret,
+  KubernetesSecretType
+}
+import org.broadinstitute.dsde.workbench.google2.KubernetesSerializableName.NamespaceName
+import org.broadinstitute.dsde.workbench.leonardo.config._
+import org.broadinstitute.dsde.workbench.leonardo.db.{
+  DbReference,
+  appErrorQuery,
+  kubernetesClusterQuery,
+  nodepoolQuery,
+  _
+}
+import org.broadinstitute.dsde.workbench.leonardo.http._
+import org.broadinstitute.dsde.workbench.leonardo.http.service.AppNotFoundException
+import org.broadinstitute.dsde.workbench.leonardo.monitor.LeoPubsubMessage.CreateAppMessage
+import org.broadinstitute.dsde.workbench.leonardo.util.{SetUpProjectNetworkParams, VPCAlgebra}
+
+import scala.collection.JavaConverters._
+import scala.concurrent.ExecutionContext
+
+final case class GKEInterpreterConfig(securityFiles: SecurityFilesConfig,
+                                      ingressConfig: IngressHelmConfig,
+                                      monitorConfig: AppMonitorConfig,
+                                      clusterConfig: KubernetesClusterConfig)
+class GKEInterpreter[F[_]: Parallel: ContextShift](
+  config: GKEInterpreterConfig,
+  vpcAlg: VPCAlgebra[F],
+  gkeService: org.broadinstitute.dsde.workbench.google2.GKEService[F],
+  kubeService: org.broadinstitute.dsde.workbench.google2.KubernetesService[F],
+  blocker: Blocker
+)(implicit val executionContext: ExecutionContext,
+  timer: Timer[F],
+  logger: StructuredLogger[F],
+  dbRef: DbReference[F],
+  F: ConcurrentEffect[F]) {
+
+  def createApp(msg: CreateAppMessage)(
+    implicit ev: ApplicativeAsk[F, AppContext]
+  ): F[Unit] = {
+    val createApp = for {
+      ctx <- ev.ask
+      _ <- logger.info(s"beginning app creation for app ${msg.appId} | trace id : ${ctx.traceId}")
+      getAppOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(msg.project, msg.appName).transaction
+      getApp <- F.fromOption(getAppOpt, AppNotFoundException(msg.project, msg.appName, ctx.traceId))
+      gkeClusterId = getApp.cluster.getGkeClusterId
+      namespaceName = getApp.app.appResources.namespace.name
+
+      _ <- kubeService.createNamespace(gkeClusterId, KubernetesNamespace(namespaceName))
```

Comment:
Wondering if there should be a check here, to not create the namespace/secrets if they already exist in the cluster. It should only be needed the first time the user deploys an app.

Reply:
I'm also wondering if in the future we should bake the namespace/secret creation into a `leo-helm` chart which we can deploy alongside the app helm chart. That might take care of the "create-if-it-doesn't-exist" semantics.

Reply:
I think we should aim to delete these resources on deleteApp, so any duplicate resources should be taken care of in front leo when we detect the app already exists